packages/framework/ai-collab/src/test/explicit-strategy/agentEditing.spec.ts

```diff
@@ -0,0 +1,271 @@
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "Vector1",
+				x: 1,
+				y: 2,
+			}),
+		);
+	});
+
+	it("handles nested objects", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: RootObject }));
+		view.initialize({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "RootObject1",
+				str: "hello",
+				vectors: [
+					{
+						[objectIdKey]: "Vector1",
+						x: 1,
+						y: 2,
+						z: 3,
+					},
+				],
+				bools: [true],
+			}),
+		);
+
+		assert.equal(idGenerator.getNode("RootObject1"), view.root);
+		assert.equal(idGenerator.getNode("Vector1"), view.root.vectors.at(0));
+
+		// const hydratedObject = hydrate(
+		// 	RootObject,
+		// 	new RootObject({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] }),
+		// );
+		// assert.equal(
+		// 	toDecoratedJson(idGenerator, hydratedObject),
+		// 	JSON.stringify({
+		// 		[objectIdKey]: "RootObject0",
+		// 		str: "hello",
+		// 		vectors: [
+		// 			{
+		// 				[objectIdKey]: "Vector0",
+		// 				x: 1,
+		// 				y: 2,
+		// 				z: 3,
+		// 			},
+		// 		],
+		// 		bools: [true],
+		// 	}),
+		// );
+		// assert.equal(idGenerator.getNode("RootObject0"), hydratedObject);
+		// assert.equal(idGenerator.getNode("Vector0"), hydratedObject.vectors.at(0));
+	});
+
+	it("handles non-POJO mode arrays", () => {
+		const sf = new SchemaFactory("testSchema");
+		class NamedArray extends sf.array("Vector", sf.number) {}
+		class Root extends sf.object("Root", {
+			arr: NamedArray,
+		}) {}
+
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: Root }));
+		view.initialize({ arr: [1, 2, 3] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({ __fluid_objectId: "Root1", arr: [1, 2, 3] }),
+		);
+
+		// const hydratedObject = hydrate(Root, new Root({ arr: [1, 2, 3] }));
+		// assert.equal(
+		// 	toDecoratedJson(idGenerator, hydratedObject),
+		// 	JSON.stringify({ __fluid_objectId: "Root0", arr: [1, 2, 3] }),
+		// );
+	});
+});
+
+describe("Makes TS type strings from schema", () => {
+	it("for objects with primitive fields", () => {
+		const testSf = new SchemaFactory("test");
+		class Foo extends testSf.object("Foo", {
+			x: testSf.number,
+			y: testSf.string,
+			z: testSf.optional(testSf.null),
+		}) {}
+		assert.equal(
+			getPromptFriendlyTreeSchema(getJsonSchema(Foo)),
+			"interface Foo { x: number; y: string; z: null | undefined; }",
+		);
+	});
+
+	// This test fails due to the fact that identifier fields are incorrectly set as optional in the JSON Schema
```

Comment:
Is this something we plan to address? An `AB#<id>` reference to a tracking item would be good.